### DISCUSSION TITLE
[Feature Request] Switch Pro Controller as Gamepad Type

### DISCUSSION THREAD

**Discussion by josh-the-tech** [LizardByte/community#603] [Sunshine] [answered]
### Select Topic Area

Feature Request

### Body

Hi,

I'm using a Switch Pro Controller with Moonlight (and Sunshine on the host PC) and when playing games it doesn't map the controls natively as Nintendo have the ABXY buttons the other way round.

Would it be possible to add Switch Controller as a controller type so these are natively mapped correctly without needing to change the mappings in each game depending on the controller?

Hope this is something that can be done. Thanks so much for all the work and effort put into this!

Josh

**Comment by ReenigneArcher** [comment]
This is available on Linux already (in beta/pre-release). On Windows we rely on vigembus, which only supports DS4 and Xbox 360.

**Reply by yemregul94** [reply]
vigembus discontinued so should we give up hope on this matter

**Reply by ReenigneArcher** [reply]
Something needs to happen on Windows side for this to be added.

A. Someone starts maintaining vigembus
B. Someone creates a free and open source alternative to vigembus
C. We start maintaining vigembus

All options have the same difficult requirement, which is the driver requires a EV signing certificate. That is complicated because it requires an official business entity to license the certification. Not the easiest thing for a small open source project run by volunteers, and who wants to subject themselves to more complicated tax filings just because of a hobby project?

**Comment by ReenigneArcher** [comment, marked as answer]
Added to Windows in https://github.com/LizardByte/Sunshine/pull/5368
